security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala

```diff
@@ -53,7 +58,23 @@ object OAuth2ErrorHandler {
     override def getMessage: String = errors.toList.mkString("OAuth2 exception with the following errors:\n - ", "\n - ", "")
   }
 
-  case class OAuth2JwtError(msg: String) extends OAuth2Error
+  trait OAuth2JwtError extends OAuth2Error
+
+  case class OAuth2JwtDecodeRawError(rawToken: String, cause: Throwable) extends OAuth2JwtError {
+    override def msg: String = s"Failure in jwt decode: ${cause.getLocalizedMessage}. Token: $rawToken"
+  }
+
+  case class OAuth2JwtKeyDetermineError(token: ParsedToken, cause: Throwable) extends OAuth2JwtError {
+    override def msg: String = s"Failure in key determining: ${cause.getLocalizedMessage}. Token: $token"
+  }
+
+  case class OAuth2JwtDecodeClaimsError(token: ParsedToken, publicKey: PublicKey, cause: Throwable) extends OAuth2JwtError {
+    override def msg: String = s"Failure in decoding json using public key: ${cause.getLocalizedMessage}. Public key: $publicKey, token: $token"
```

Comment:
token + claims are sth that is safe to log?

Reply:
You've right that leakage of token or claims would be risky, I'll obfuscate that things

Reply:
I will change it in next PR to avoid conflicts